### PR TITLE
feat: support likes on projects & reports

### PR DESCRIPTION
**Pull request by cabreraalex**
# Description
Fix ZEN-245

Screenshot 2023-10-19 at 13 31 24

### CODE REVIEW

File: frontend/src/lib/components/project/Project.svelte [hunk 1]
```diff
@@ -144,14 +145,42 @@
...
+					}}
+				>
+					<Tooltip content={`Like this project!`} theme={'zeno-tooltip'} position="bottom">
+						<Icon tag="svg" viewBox="0 0 24 24">
+							<path d={stats.userLiked ? mdiHeart : mdiHeartOutline} />
+						</Icon>
+					</Tooltip>
+				</button>
+			{:else}
+				<div class=" w-6 h-6 fill-primary cursor-not-allowed">
```
Comment (Sparkier) [inline frontend/src/lib/components/project/Project.svelte:172]: could redirect to login on click

Comment (Sparkier) [reply frontend/src/lib/components/project/Project.svelte:172]: if so make it a button

File: frontend/src/lib/components/report/Report.svelte [hunk 2]
```diff
@@ -111,5 +112,42 @@
...
+					}}
+				>
+					<Tooltip content={`Like this report!`} theme={'zeno-tooltip'} position="bottom">
+						<Icon tag="svg" viewBox="0 0 24 24">
+							<path d={stats.userLiked ? mdiHeart : mdiHeartOutline} />
+						</Icon>
+					</Tooltip>
+				</button>
+			{:else}
+				<div class=" w-6 h-6 fill-primary cursor-not-allowed">
```
Comment (Sparkier) [inline frontend/src/lib/components/report/Report.svelte:139]: same here

File: frontend/src/routes/(app)/(home)/[user]/reports/+page.svelte [hunk 3]
```diff
@@ -33,9 +33,14 @@
 {/if}
 <div class="flex flex-wrap mb-6 h-full content-start">
 	{#each ownReports as report}
-		<Report report={report.report} stats={report.statistics} deletable />
+		<Report
+			report={report.report}
+			stats={report.statistics}
+			loggedIn={data.user !== null}
```
Comment (Sparkier) [inline frontend/src/routes/(app)/(home)/[user]/reports/+page.svelte:39]: They have to be logged in to see this page, no?

File: frontend/src/routes/(app)/(home)/[user]/reports/+page.svelte [hunk 4]
```diff
@@ -33,9 +33,14 @@
...
+			deletable
+		/>
 	{/each}
 	{#each sharedReports as report}
-		<Report report={report.report} stats={report.statistics} />
+		<Report report={report.report} stats={report.statistics} loggedIn={data.user !== null} />
```
Comment (Sparkier) [inline frontend/src/routes/(app)/(home)/[user]/reports/+page.svelte:44]: Same, don't they have to be logged in here?

**Review by Sparkier — APPROVED** [review]

File: frontend/src/lib/components/project/Project.svelte (hunk 1 above)
Comment (cabreraalex) [reply frontend/src/lib/components/project/Project.svelte:172]: good idea

**Pull request merged**